### ISSUE TITLE
Addon Language Filter - Option to completely hide a language that is not spoken

### ISSUE THREAD

**Issue by Singh-in** [friendica/friendica#10136]
"I always get frustrated when I see a language in the stream that I don't speak. I would like to hide this completely.

### Describe the function you want.

The addon "Language Filter" detects, after a certain probability, posts by language and offers the possibility to reduce these posts. However, these posts are not completely hidden so that they can still be viewed if necessary. 

Unfortunately, there is no option to completely hide posts in languages that are not spoken, so that only the content is displayed whose language is actually spoken. 

As a result, it often happens that only reduced contributions are displayed on the community page. Contributions in the spoken language are hidden behind tons of reduced contributions. This makes the community page less useful.

It would therefore be a great feature if you could optionally filter (remove) all these posts completely and only receive the content whose language you speak.

**Comment by MrPetovan** [comment]
Technical note: the langfilter addon piggy-backs off the content filter feature that only collapses posts. This request would require altering the community timeline post database query which is way more involved.

**Comment by Singh-in** [comment]
#10052

**Comment by bkil** [comment]
I'd like to support this feature request from a user perspective to easy life on multi-national instances and I did consider requesting this in the past.

However, note that language detection can make errors quite often, especially if the post has an insufficient amount of text. If no placeholder appears in your timeline, how can you possibly override the decision of the machine?

The following edge cases also show how difficult this is:
* An external link that may or may not point to content in a language that you speak (some sites also allow switching between languages manually or automatically)
* An image that may or not not need any language comprehension at all (or where the few words burned into it aren't necessary for comprehension)
* A video that may or may not contain audio tracks or subtitles in a language you understand

I don't have a good recommendation as of now, but I'm all ears. All workarounds I could think of are a bit of a kludge:
* Better enforcement of specifying a language with new posts, perhaps even showing what the machine detects to motivate the user in correcting it
* Collaboratively tagging the language of each other's posts (maybe introducing a new language tag for "culture and language independent")
* Restrict detection error by intersecting the language detector's output of one's posts by the user's language (that may also be detected or tagged collaboratively)
* Just join a local instance where others are speaking a language that you understand and ask others to post in a single language using any given single account (and perhaps create multiple accounts)

**Comment by schmaker** [comment]
Well, even though lang-detection addon is not perfect, i seriously would like to support this idea. Even hiding posts with Russian and Chinese / Japanese text would clean the mess a lot.

**Comment by bkil** [comment]
> Well, even though lang-detection addon is not perfect, i seriously would like to support this idea. Even hiding posts with Russian and Chinese / Japanese text would clean the mess a lot.

- `( ͡° ͜ʖ ͡°)`
- `¯\_(ツ)_/¯`

**Comment by MrPetovan** [comment]
Good idea, but the Language Detection library is using a weighted approach so these wouldn't be ruled as 100% non-Latin languages just based on these smileys.

**Comment by bkil** [comment]
Yes, a single emoji would fall through the threshold of the detector - it would need much more text to decide. I just wanted to warn you to exercise caution before doing a Unicode code point based grep blindly.
